src/internal/m365/collection/drive/delta_tree.go

```diff
@@ -0,0 +1,80 @@
+package drive
+
+import (
+	"time"
+
+	"github.com/alcionai/corso/src/pkg/path"
+)
+
+// folderyMcFolderFace owns our delta processing tree.
+type folderyMcFolderFace struct {
+	// tenant/service/resource/category/driveID
+	// (or whatever variant the service defines)
+	// allows the tree to focus only on folder structure,
+	// and minimizes the possibility of multi-prefix path bugs.
+	prefix path.Path
+
+	// the root of the tree;
+	// new, moved, and notMoved collections
+	collections *nodeyMcNodeFace
+
+	// the majority of operations we perform can be handled with
+	// a folder id lookup instead of re-walking the entire tree.
```

Comment:
nit: just capitalization
```suggestion
	// a folder ID lookup instead of re-walking the entire tree.
```